lib/drivers/abbottFreeStyle.js

```diff
@@ -0,0 +1,562 @@
+      handler: function(match, packet) {
+        var entry = {};
+        var reading = match[1];
+        if (reading == 'HI ') {
+          entry.reading = 501;
+        } else if (reading == 'LO ') {
+          entry.reading = 20;
+        } else {
+          entry.reading = parseInt(reading, 10);
+        }
+        entry.datetime = parseDateTime(match, 2);
+        var readingType = match[8];
+        if (readingType === 'K') {
+          entry.readingType = 'ketone';
+        } else {
+          entry.readingType = 'glucose';
+        }
+        entry.status = parseInt(match[9], 16);   // this is in hex
+        packet.logEntries.push(entry);
+      }
+    },
+    checksum: {
+      pattern: '^(0x[0-9A-F]{4})  END',
+      handler: function(match, packet) {
+        packet.receivedChecksum = parseInt(match[1], 16); // in hex
+      }
+    }
+  };
+  var logEntry = 'logEntry';
+
+
+  // accepts a stream of bytes and tries to find a packet
+  // at the beginning of it.
+  // returns a packet object; if valid == true it's a valid packet
+  // if packet_len is nonzero, that much should be deleted from the stream
+  // if valid is false and packet_len is nonzero, the previous packet
+  // should be NAKed.
+  // don't call this if you don't have at least a few bytes available
+  var extractPacket = function (bytes) {
+    var packet = {
+      bytes: bytes,
+      valid: false,
+      packet_len: 0,
+      lcb: 0,
+      lines: [],
+      logEntries: []
+    };
+
+    // look for a terminating ETX
+    var startIndex = 0;
+    var endIndex = _.indexOf(bytes, ETX, startIndex + 1);
+    if (endIndex === -1) {
+      // didn't find end of packet, so bail for now
+      return packet;
+    }
+
+    // we have what looks like a packet, so now's a good time to
+    // calculate the checksum if we have one
+    // first, we have to find the start of the checksum, if it exists.
+    // let's convert the whole array to strings
+
+    // I tried to do this with:
+    // _.map(bytes, String.fromCharCode).join('');
+    // but I kept getting semi-random characters in the resulting string, so I'm using this instead
+    function stringFromBytes(bytes, start, end) {
+      var s = '';
+      for (var i=start; i<end; ++i) {
+        s += String.fromCharCode(bytes[i]);
+      }
+      return s;
+    }
+
+    // this is a search for a fixed set of bytes within a larger block. It's not exactly
+    // Boyer-Moore, but it'll do for now
+    function searchForSequence(bytes, sequence) {
+      var found = false;
+      for (var ix=0; ix<bytes.length; ++ix) {
+        found = true;
+        for (var jx=0; jx<sequence.length; ++jx) {
+          if (bytes[ix + jx] != sequence[jx]) {
+            found = false;
+            break;
+          }
+        }
+        if (found) {
+          return ix;
+        }
+      }
+      return -1;
+    }
+
+    // bstring = stringFromBytes(bytes, 0, bytes.length);
+    // console.log(bstring);
```

Comment:
Do you anticipate needing this (and any other) logging statements in the near future or can they be deleted?